test/saithrift/src/saiserver.cpp

```diff
@@ -47,7 +49,80 @@ void on_switch_state_change(_In_ sai_object_id_t switch_id,
 void on_fdb_event(_In_ uint32_t count,
                   _In_ sai_fdb_event_notification_data_t *data)
 {
-}
+    sai_fdb_event_t event_type;
+    sai_fdb_entry_t fdb_entry;
+    uint32_t attr_count;
+    sai_attribute_t *attr;
+    sai_object_id_t bv_id;	
+    sai_object_id_t bport_id;
+	
+    attr = data->attr;
+    event_type = data->event_type;
+    fdb_entry = data->fdb_entry;
+    bv_id = fdb_entry.bv_id;
+    attr_count = data ->attr_count;
+    
+    for (uint32_t i = 0; i < attr_count; i++)
+    {
+        if(attr[i].id == SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID)
+            bport_id = attr[i].value.oid;
+    }
+           
+    sai_fdb_entry_t fdb_m;
+    sai_object_id_t b_id;
+      
+    switch(event_type)
+    {   
+        case SAI_FDB_EVENT_LEARNED
+            gFdbMap.insert(std::pair<std::sai_fdb_entry_t,std::sai_fdb_event_t>(fdb_entry,bport_id));  
+            break;
+            
+        case SAI_FDB_EVENT_FLUSHED: 
+            if ( bv_id == 0 && bport_id == 0 )
+                gFdbMap.clear;
+            
+            else
+            {    
+                for(auto it = gFdbMap.begin(); it != gFdbMap.end(); it++)
+                {   
+                    fdb_m = it->first;                  
+                    b_id = it->second; 				
+	                
+                    if(bport_id == 0 && bv_id == fdb_m.bv_id)
+                        it = gFdbMap.erase(it);
+                    else if(bv_id == 0 && bport_id == b_id)		
+                        it = gFdbMap.erase(it);
+                    else if(bv_id == fdb_m.bv_id && bport_id == b_id)
+                        it = gFdbMap.erase(it);
+                }
+            }
+            break;
+	      
+        case SAI_FDB_EVENT_MOVE:
+            for(auto it = gFdbMap.begin(); it != gFdbMap.end(); it++)
+            {
+                fdb_m = it->first;
+                
+                if(fdb_entry.mac_address == fdb_m.mac_address && bv_id == fdb_m.bv_id)
+                    it->second = bport_id;
+            }
+            break;
+        
+        case SAI_FDB_EVENT_AGED:
+            for(auto it = gFdbMap.begin(); it != gFdbMap.end(); it++)
+            {
+                fdb_m = it->first;
+                
+                if(fdb_entry.mac_address == fdb_m.mac_address && bv_id == fdb_m.bv_id)
```

Comment:
How about bport_id. Where do you match that since the aging can be port specific.